### PR TITLE
support event listener

### PR DESCRIPTION
**Pull request by BusyJay**
Only some event callbacks are implemented, others can be implemented when necessary.

### CODE REVIEW

File: librocksdb_sys/src/lib.rs [hunk 1]
```diff
@@ -981,6 +987,59 @@ extern "C" {
...
+                                                     pos: size_t,
+                                                     len: *mut size_t)
+                                                     -> *const c_char;
+    pub fn crocksdb_compactionjobinfo_table_properties(info: *const DBCompactionJobInfo)
+                                                       -> *const DBTablePropertiesCollection;
+
+    pub fn crocksdb_externalfileingestioninfo_cf_name(info: *const DBIngestionInfo,
+                                                      size: *mut size_t)
+                                                      -> *const c_char;
+    #[cfg_attr(rustfmt, rustfmt_skip)]
```
Comment (huachaohuang) [inline librocksdb_sys/src/lib.rs]: What's the problem of rustfmt here?

File: librocksdb_sys/crocksdb/c.cc [hunk 2]
```diff
@@ -1541,6 +1555,145 @@ size_t crocksdb_options_get_block_cache_usage(crocksdb_options_t *opt) {
...
+crocksdb_eventlistener_t* crocksdb_eventlistener_create(
+    void* state_, void (*destructor_)(void*),
+    void (*on_flush_completed)(void*, crocksdb_t*,
+                               const crocksdb_flushjobinfo_t*),
+    void (*on_compaction_completed)(void*, crocksdb_t*,
+                                    const crocksdb_compactionjobinfo_t*),
+    void (*on_external_file_ingested)(
+        void*, crocksdb_t*, const crocksdb_externalfileingestioninfo_t*)) {
+  crocksdb_eventlistener_t* et = new crocksdb_eventlistener_t;
+  et->state_ = state_, et->destructor_ = destructor_;
```
Comment (huachaohuang) [inline librocksdb_sys/crocksdb/c.cc]: Not good for eyes, please split to two line.

**Comment by zhangjinpeng1987** [comment]
LGTM

File: librocksdb_sys/crocksdb/c.cc [hunk 3]
```diff
@@ -1541,6 +1555,146 @@ size_t crocksdb_options_get_block_cache_usage(crocksdb_options_t *opt) {
...
+};
+
+crocksdb_eventlistener_t* crocksdb_eventlistener_create(
+    void* state_, void (*destructor_)(void*),
+    void (*on_flush_completed)(void*, crocksdb_t*,
+                               const crocksdb_flushjobinfo_t*),
+    void (*on_compaction_completed)(void*, crocksdb_t*,
+                                    const crocksdb_compactionjobinfo_t*),
+    void (*on_external_file_ingested)(
+        void*, crocksdb_t*, const crocksdb_externalfileingestioninfo_t*)) {
```
Comment (disksing) [inline librocksdb_sys/crocksdb/c.cc]: Too hard to read. maybe use some typedef for the functions?

**Pull request merged**